File: 코딩 기초 트레이닝/문자 개수 세기-p181902/문자 개수 세기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181902

function solution(my_string) {
    // # 0 ~ 25: A ~ Z
    // # 26 ~ 52: a ~ z
    // # A ~ Z: 65 ~ 90
    // # a ~ z: 97 ~ 122
    const answer = new Array(52).fill(0);

    my_string.split('').forEach((s) => {
        if(s.charCodeAt() <= 90) {
            answer[s.charCodeAt() % 65] += 1;
        } else {
            answer[s.charCodeAt() % 97 + 26] += 1;
        }
    });


    return answer;
}

File: 코딩 기초 트레이닝/문자 개수 세기-p181902/문자 개수 세기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181902

def solution(my_string):
    answer = [0 for i in range(52)]
    # 0 ~ 25: A ~ Z
    # 26 ~ 52: a ~ z
    # A ~ Z: 65 ~ 90
    # a ~ z: 97 ~ 122

    for s in my_string:
        if ord(s) <= 90:
            answer[ord(s)%65] += 1
        else:
            answer[ord(s)%97 + 26] += 1

    return answer
